framework/wazuh/core/configuration.py

```diff
@@ -798,62 +795,129 @@ def get_active_configuration(agent_id, component, configuration):
+                                'logcollector': 'logcollector', 'mail': 'mail', 'monitor': 'monitor',
+                                'request': 'remote', 'syscheck': 'syscheck', 'wmodules': 'wmodules'}
+
     if not component or not configuration:
         raise WazuhError(1307)
 
-    components = {"agent", "agentless", "analysis", "auth", "com", "csyslog", "integrator", "logcollector", "mail",
-                  "monitor", "request", "syscheck", "wmodules"}
-
-    # checks if the component is correct
+    # Check if the component is correct
+    components = component_socket_mapping.keys()
     if component not in components:
         raise WazuhError(1101, f'Valid components: {", ".join(components)}')
 
-    sockets_path = os_path.join(common.WAZUH_PATH, "queue", "sockets")
+    def get_active_configuration_000():
+        """Get manager active configuration."""
+        # Communicate with the socket that corresponds to the component requested
+        dest_socket = os_path.join(common.WAZUH_PATH, "queue", "sockets", component_socket_mapping[component])
 
-    if agent_id == '000':
-        dest_socket = os_path.join(sockets_path, component)
-        command = f"getconfig {configuration}"
         # Verify component configuration
         if not os.path.exists(dest_socket):
-            raise WazuhError(1121, extra_message=f"please verify that the component '{component}' \
-                is properly configured")
-    else:
-        dest_socket = os_path.join(sockets_path, "request")
-        command = f"{str(agent_id).zfill(3)} {component} getconfig {configuration}"
+            raise WazuhError(1121,
+                             extra_message=f"please verify that the component '{component}' is properly configured")
 
-    # Socket connection
-    try:
-        s = WazuhSocket(dest_socket)
-    except Exception:
-        raise WazuhInternalError(1121)
+        # Simple socket message
+        if component_socket_mapping[component] not in sockets_json_protocol:
+            msg = f"{GETCONFIG_COMMAND} {configuration}"
 
-    # Send message
-    s.send(command.encode())
+            # Socket connection
+            try:
+                s = wazuh_socket.WazuhSocket(dest_socket)
+            except Exception:
+                raise WazuhInternalError(1121)
 
-    # Receive response
-    try:
-        # Receive data length
-        rec_msg_ok, rec_msg = s.receive().decode().split(" ", 1)
-    except ValueError:
-        raise WazuhInternalError(1118, extra_message="Data could not be received")
+            # Send message
+            s.send(msg.encode())
+
+            # Receive response
+            try:
+                # Receive data length
+                rec_msg_ok, rec_msg = s.receive().decode().split(" ", 1)
+            except ValueError:
+                raise WazuhInternalError(1118, extra_message="Data could not be received")
+
+            s.close()
```

Comment:
Either call the `close()` method in a `finally` block or use the context manager to guarantee that the socket will close.

Reply:
`finally` added